### PR TITLE
feat(sgcloudspanner): add RunEmulator helper

### PR DESCRIPTION
**Pull request by odsod**
Starts a dockerized Cloud Spanner emulator and sets the
SPANNER_EMULATOR_HOST environment variable.

### CODE REVIEW

**Review by Edholm — APPROVED** [review]
Nice! Excellent as always :godmode:

File: tools/sgcloudspanner/emulator.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	} else {
+		dockerRunCmd = sg.Command(ctx, "docker", "run", "-d", "--publish-all", image)
+	}
+	var dockerRunStdout strings.Builder
+	dockerRunCmd.Stdout = &dockerRunStdout
+	if err := dockerRunCmd.Run(); err != nil {
+		return nil, err
+	}
+	containerID := strings.TrimSpace(dockerRunStdout.String())
+	cleanup := func() {
```
Comment (sfinnman) [inline tools/sgcloudspanner/emulator.go:53]: Could we defer cleanups and do an if check on the err named return?

Comment (odsod) [reply tools/sgcloudspanner/emulator.go:53]: I think the readability benefit would be marginal - would prefer to keep as is

Comment (sfinnman) [reply tools/sgcloudspanner/emulator.go:53]: Alright! Still a little green on go conventions!

**Review by sfinnman — APPROVED** [review]
LGTM

**Pull request merged**